src/charm.py

```diff
@@ -120,14 +120,17 @@ def _configure(self, _):
         # Restart prometheus only if command line arguments have changed,
         # otherwise just reload its configuration.
         if current_services == new_layer.services:
-            self._prometheus_server.reload_configuration()
-            logger.info("Prometheus configuration reloaded")
+            reloaded = self._prometheus_server.reload_configuration()
```

Comment:
walrus?

Reply:
Actually in this case even walrus is not necessary. On can just write `if not self._prometheus_server.reload_configuration()`. Is that easier to understand than `if not reloaded` ? Which of is clearer (more like natural language) to read ?

Reply:
Yes I see you do not use the value of `reloaded` inside. In that case I would swap the `if` to be positive:
```python
if self._prometheus_server.reload_configuration():
    self.unit.status = ActiveStatus()
    # etc
else:
    self.unit.status = BlockedStatus("Failed to load Prometheus config")
```

Reply:
Even then is it nicer to read `if reloaded` or `if self._prometheus_server.reload_configuration()` ?